src/SectorView.cpp

```diff
@@ -585,48 +586,60 @@ std::vector<SystemPath> SectorView::GetRoute()
 
-	int sec_dist = ceilf(dist / Sector::SIZE);
-
-	std::vector<SystemPath> nodes;
-	std::set<std::vector<SystemPath>::size_type> unvisited;
-
-	std::vector<float> path_dist; // distance from source to node
-	std::vector<std::vector<SystemPath>::size_type> path_prev; // previous node in optimal path
+	const int sec_dist = ceilf(dist / Sector::SIZE);
 
 	// nodes[0] is always start
+	std::vector<SystemPath> nodes;
 	nodes.push_back(start);
 
+	const Sint32 minX = std::min(start.sectorX, target.sectorX)-2, maxX = std::max(start.sectorX, target.sectorX)+2;
+	const Sint32 minY = std::min(start.sectorY, target.sectorY)-2, maxY = std::max(start.sectorY, target.sectorY)+2;
+	const Sint32 minZ = std::min(start.sectorZ, target.sectorZ)-2, maxZ = std::max(start.sectorZ, target.sectorZ)+2;
+	const vector3f start_pos = start_sec->m_systems[start.systemIndex].GetFullPosition();
+	const vector3f target_pos = target_sec->m_systems[target.systemIndex].GetFullPosition();
+
 	// go sector by sector for sec_dist sectors and add systems
 	// if they are within 110% of dist of both start and target
-	for (Sint32 sx = -sec_dist; sx <= sec_dist; sx++) {
-		for (Sint32 sy = -sec_dist; sy <= sec_dist; sy++) {
-			for (Sint32 sz = -sec_dist; sz < sec_dist; sz++) {
-				SystemPath sec_path = SystemPath(start.sectorX + sx, start.sectorY + sy, start.sectorZ + sz);
+	for (Sint32 sx = minX; sx <= maxX; sx++) {
+		for (Sint32 sy = minY; sy <= maxY; sy++) {
+			for (Sint32 sz = minZ; sz < maxZ; sz++) {
+				const SystemPath sec_path = SystemPath(sx, sy, sz);
 				RefCountedPtr<const Sector> sec = m_galaxy->GetSector(sec_path);
 				for (std::vector<Sector::System>::size_type s = 0; s < sec->m_systems.size(); s++) {
-					if (start.IsSameSystem(sec->m_systems[s].GetPath())) continue; // start is already nodes[0]
+					if (start.IsSameSystem(sec->m_systems[s].GetPath()))
+						continue; // start is already nodes[0]
+
+					bool isWithinLineSegment = false;
+					const float lineDist = MathUtil::DistanceFromLine(start_pos, target_pos, sec->m_systems[s].GetFullPosition(), isWithinLineSegment);
+
 					if (Sector::DistanceBetween(start_sec, start.systemIndex, sec, sec->m_systems[s].idx) <= dist * 1.10 &&
-						Sector::DistanceBetween(target_sec, target.systemIndex, sec, sec->m_systems[s].idx) <= dist * 1.10) {
+						Sector::DistanceBetween(target_sec, target.systemIndex, sec, sec->m_systems[s].idx) <= dist * 1.10 &&
+						/*isWithinLineSegment &&*/ lineDist<(Sector::SIZE*3))
+					{
 						nodes.push_back(sec->m_systems[s].GetPath());
 					}
 				}
 			}
 		}
 	}
+	Output("SectorView::AutoRoute, nodes to search = %lu\n", nodes.size());
 
 	// setup inital values and set everything as unvisited
+	std::vector<float> path_dist; // distance from source to node
+	std::vector<std::vector<SystemPath>::size_type> path_prev; // previous node in optimal path
+	std::set<std::vector<SystemPath>::size_type> unvisited;
```

Comment:
std::set operations while being O(log N), are often slow because of memory allocations.
If we don't need the order (and it doesn't seem that we do), unordered_set will perform better.
Furthermore, as we don't need to check whether something is in the set, but only iterate over it, I'd replace that std::set with std::vector (and when element is erased, just copy the last element of a vector to a "hole")